### PR TITLE
Feature/user ranking

### PR DESCRIPTION
**Pull request by PotSchmoell**

### CODE REVIEW

File: lambda_layers/core_layer/python/core_layer/alembic/versions/35997fffdb9d_add_finish_timestamps_to_closed_reviews.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+"""Add finish timestamps to closed reviews
+
+Revision ID: 35997fffdb9d
+Revises: 7e1207d1a7e1
```
Comment (SandroGlueck) [inline lambda_layers/core_layer/python/core_layer/alembic/versions/35997fffdb9d_add_finish_timestamps_to_closed_reviews.py:4]: Is it a good idea to have two revisions with the same `down_revision` this one and item_types?

File: lambda_functions/user_service/get_user.py [hunk 2]
```diff
@@ -24,7 +24,21 @@ def get_user(event, context, is_test=False, session=None):
...
                 user, is_test, session)
+            total_rank = user_handler.get_user_rank(
+                user, False, is_test, session)
+            level_rank = user_handler.get_user_rank(
+                user, True, is_test, session)
+            solved_cases_total = user_handler.get_solved_cases(
+                user, False, is_test, session)
+            solved_cases_today = user_handler.get_solved_cases(
+                user, True, is_test, session)
+            exp_needed = user_handler.get_needed_exp(user, is_test, session)
```
Comment (SandroGlueck) [inline lambda_functions/user_service/get_user.py:35]: Wouldn't it be possible to get all data in one DB query?

File: lambda_layers/core_layer/python/core_layer/helper.py [hunk 3]
```diff
@@ -5,15 +5,16 @@
...
 
 
-def get_date_time_now(is_test):
-    if is_test:
-        return datetime.now()
-    else:
-        return datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+def get_date_time_now(is_test=True):
+    return func.now()
+    # TODO: Remove this method. Use func.now() and set create_time in model
```
Comment (SandroGlueck) [inline lambda_layers/core_layer/python/core_layer/helper.py:15]: 👍

File: lambda_functions/user_service/get_user.py (hunk 2 above)
Comment (PotSchmoell) [reply lambda_functions/user_service/get_user.py:35]: Maybe, but I wouldnt really see the point as this query would be incredibly complicated. We could combine the methods and return one dict, but that would make the methods less reusable.

File: lambda_layers/core_layer/python/core_layer/alembic/versions/35997fffdb9d_add_finish_timestamps_to_closed_reviews.py (hunk 1 above)
Comment (PotSchmoell) [reply lambda_layers/core_layer/python/core_layer/alembic/versions/35997fffdb9d_add_finish_timestamps_to_closed_reviews.py:4]: No https://stackoverflow.com/questions/22342643/alembic-revision-multiple-heads-due-branching-error
Maybe you can move this statement/migration to your branch?

**Review by SandroGlueck — APPROVED** [review]

File: lambda_functions/user_service/get_user.py (hunk 2 above)
Comment (PotSchmoell) [reply lambda_functions/user_service/get_user.py:35]: If we want to reduce the number of queries, I would rather create a view that extends the user table. But for now I would leave it as it is.

**Pull request merged**